ape_safe/client.py

```diff
@@ -165,72 +246,128 @@ def _all_transactions(self) -> Iterator[dict]:
-        next_nonce = self.get_next_nonce()
 
-        for txn in self._all_transactions():
-            if txn["nonce"] < starting_nonce:
-                break  # NOTE: order is largest nonce to smallest, so safe to break here
-
-            isConfirmed = len(txn["confirmations"]) >= txn["confirmationsRequired"]
-            isExecuted = "isExecuted" in txn and txn["isExecuted"]
+            for txn in data["results"]:
+                if "isExecuted" in txn and txn["isExecuted"]:
+                    yield ExecutedTxData.parse_obj(txn)
 
-            if confirmed is not None:
-                if not confirmed and isExecuted:
-                    break  # NOTE: Break at the first executed transaction
+                else:
+                    yield UnexecutedTxData.parse_obj(txn)
 
-                elif confirmed and not isConfirmed:
-                    continue  # NOTE: Skip not confirmed transactions
-
-            if txn["nonce"] < next_nonce and not isExecuted:
-                continue  # NOTE: Skip orphaned transactions
-
-            if filter_by_ids and txn["safeTxHash"] not in filter_by_ids:
-                continue  # NOTE: Skip transactions not in the filter
+            url = data["next"]
 
-            if filter_by_missing_signers and filter_by_missing_signers.issubset(
-                set(conf["owner"] for conf in txn["confirmations"])
-            ):
-                # NOTE: Skip if all signers from `filter_by_missing_signers`
-                #       are in `txn.confirmations`
-                continue
+    def get_confirmations(self, safe_tx_hash: SafeTxID) -> Iterator[SafeTxConfirmation]:
+        url = (
+            f"{self.transaction_service_url}/api"
+            f"/v1/multisig-transactions/{str(safe_tx_hash)}/confirmations"
+        )
+        while url:
+            response = requests.get(url)
+            if not response.ok:
+                raise ClientResponseError(url, response)
 
-            yield ExecutedTxData.parse_obj(txn) if isExecuted else UnexecutedTxData.parse_obj(txn)
+            data = response.json()
+            yield from map(SafeTxConfirmation.parse_obj, data["results"])
+            url = data["next"]
 
-    def post_transaction(self, safe_tx: SafeTx, sigs: Optional[List[MessageSignature]] = None):
-        tx_data = UnexecutedTxData.from_safe_tx(safe_tx)
-        if sigs:
-            tx_data.signatures = HexBytes(
-                reduce(lambda raw_sig, next_sig: raw_sig + next_sig.encode_vrs(), sigs, b"")
+    def post_transaction(self, safe_tx: SafeTx, sigs: Dict[AddressType, MessageSignature]):
+        tx_data = UnexecutedTxData.from_safe_tx(safe_tx, self.safe_details.threshold)
+        tx_data.signatures = HexBytes(
+            reduce(
+                lambda raw_sig, next_sig: raw_sig + next_sig.encode_rsv(),
+                order_by_signer(sigs),
+                b"",
             )
+        )
 
         url = f"{self.transaction_service_url}/api/v1/multisig-transactions"
```

Comment:
This URL seems wrong?

I had to use:

I think it might be:

```
 url = f"{self.transaction_service_url}/api/v1/safes/{tx_data.safe}/multisig-transactions"
```

It seems like I kinda got that to work anyway, and this one I am not seeingin the swagger.